docs/MockFunctions.md

```diff
@@ -151,6 +151,44 @@ test('should fetch users', () => {
 });
 ```
 
+## Mocking Partials
+
+Subsets of a module can be mocked and the rest of the module can keep their actual implementation:
+
+```js
+// foo-bar-baz.js
+export const foo = 'foo';
+export const bar = () => 'bar';
+export default () => 'baz';
+```
+
+```js
+//test.js
+import defaultExport, {bar, foo} from '../foo-bar-baz';
+
+jest.mock('../foo-bar-baz', () => {
+  const originalModule = jest.requireActual('../foo-bar-baz');
+  const mockedModule = jest.createMockFromModule('../foo-bar-baz');
+
+  //Mock the default export and named export 'foo'
+  return {
+    ...mockedModule,
+    ...originalModule,
```

Comment:
Could you explain a bit why you need the `mockedModule` at all, since it's being overwritten by the `originalModule`? I don't immediately see it being helpful here, rather obfuscating the example

Reply:
Without `createMockFromModule`, default export call was throwing an error: `TypeError: (0 , _fooBarBaz.default) is not a function`. I examined this method now, I've noticed this method it is passing `__esModule: true`, which is required when using default export from an ES6 module. I removed the unnecessary usage of `createMockFromModule` and declared `__esModule: true` on mock instead @thymikee .

Reply:
Ugh, that's why I prefer using `jest.spyOn` for such cases. Maybe we could document that as well? 
Using `createMockFromModule` seems a better API than adding implementation detail like `__esModule: true`. @SimenB what do you think?

Reply:
It is already documented in https://jestjs.io/docs/jest-object#jestmockmodulename-factory-options btw.